openlcs/libs/sc_handler.py

```diff
@@ -0,0 +1,130 @@
+import koji
+import glob
+import os
+import uuid
+import sys
+
+# Fix absolute import issue in openlcs.
+openlcs_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+if openlcs_dir not in sys.path:
+    sys.path.append(openlcs_dir)
+from libs.common import compress_source_to_tarball  # noqa: E402
+
+
+class SourceContainerHandler(object):
+    """
+    Object used for handle source in source containers.
+    """
+    def __init__(self, config=None):
+        self.config = config
+
+    @staticmethod
+    def get_component_flat(data, comp_type):
+        return {
+            'uuid': str(uuid.uuid4()),
+            'type': comp_type,
+            'name': data.get('name'),
+            'version': data.get('version'),
+            'release': data.get('release'),
+            'license': '',
+            'arch': '',
+            'synced': False
+        }
+
+    def get_container_component(self, sc_nvr):
+        """
+        Get container component.
+        """
+        nvr = koji.parse_NVR(sc_nvr)
+        container_component = self.get_component_flat(nvr, 'CONTAINER_IMAGE')
+        return container_component
+
+    def get_srpm_components(self, srpm_dir, sc_nvr):
+        """
+        Get the srpm components list from source container itself. The
+        component value is almost same as the srpm component from corgi.
+        The value of srpm component from corgi is as following:
+        {
+            'uuid': 'uuid',
+            'type': 'SRPM',
+            'name': 'libcom_err',
+            'version': '1.45.6',
+            'release': '2.el8',
+            'license': '',
+            'arch': '',
+            'synced': False
+        }
+        @params: srpm_dir, destination directory for srpm files' dir
+        """
+        components = []
+        if os.path.isdir(srpm_dir):
+            srpms = glob.glob(srpm_dir + "/*.src.rpm")
+            for srpm in srpms:
+                srpm = srpm.split('/')[-1]
+                rpm_nvr = srpm.replace('.src.rpm', '')
+                nvr = koji.parse_NVR(rpm_nvr)
+                component = self.get_component_flat(nvr, 'SRPM')
+                components.append(component)
+        else:
+            err_msg = f'Not found {srpm_dir}'
+            raise RuntimeError(err_msg) from None
+        return components
+
+    def get_source_of_srpm_component(self, srpm_dir, nvr):
+        """
+        Get the source file or srpm
+        """
+        srpm_file = nvr + '.src.rpm'
+        srpm_filepath = os.path.join(srpm_dir, srpm_file)
+        if os.path.isfile(srpm_filepath):
+            return srpm_filepath
+        else:
+            err_msg = f'Not found {srpm_filepath}'
+            raise RuntimeError(err_msg) from None
```

Comment:
you are in favor of raising `RuntimeError` in a lot of places like this. But I think dealing with exceptions like this would be an overhead over time.

An alternate way is for functions that needs to get something - return the desired object when success, and return `None` in case if fails - if this is not enough, give a log or error message should be sufficient, unless it's critical enough to raise an exception and should be captured by the caller. In your implementation, I don't see how the exceptions were caught. WDYT?

Reply:
Ok, it seems None is acceptable. It's no need to raise an exception.